krb-login/src/main/java/com/odysseusinc/krblogin/KerberosServiceImpl.java

```diff
@@ -0,0 +1,281 @@
+                }
+                builder.statement("bash")
+                        .withParam("-c")
+                        .statement("echo " + dataSource.getKrbPassword() + " | " + kinitPath + KINIT_COMMAND + " " +
+                                dataSource.getKrbUser() + "@" + dataSource.getKrbRealm());
+                break;
+            case KEYTAB:
+                builder.statement(kinitPath + KINIT_COMMAND)
+                        .withParam("-k")
+                        .withParam("-t")
+                        .withParam(keytab.toString())
+                        .withParam(dataSource.getKrbUser() + "@" + dataSource.getKrbRealm());
+                break;
+            default:
+                throw new IllegalArgumentException("Unsupported authentication type");
+        }
+        String[] command = builder.build();
+        if (log.isDebugEnabled()) {
+            log.debug("Kerberos init command: {}", StringUtils.join(command, " "));
+        }
+        return command;
+    }
+
+    private Path buildTempKrbConf(DataSourceUnsecuredDTO dataSource) throws IOException {
+
+        Path configPath = Files.createTempFile("", ".conf");
+        String krbConfHeader = buildKrbConfHeader(dataSource.getKrbRealm());
+        FileUtils.write(configPath.toFile(), krbConfHeader, UTF_8);
+        extendKrbConf(configPath, dataSource);
+        return configPath;
+    }
+
+    private Path extendKrbConf(Path configPath, DataSourceUnsecuredDTO dataSource) throws IOException {
+
+        File config = configPath.toFile();
+        long fileLength = config.length();
+        try (RandomAccessFile raf = new RandomAccessFile(config, "rw")) {
+            String confStr = convertConfigToString(raf, (int) fileLength);
+
+            //remove inconsistent part (if it exists) with old values of admin_server or kdc
+            confStr = removeInconsistentRealm(dataSource, confStr);
+            boolean isRealmDefined = confStr.toLowerCase().contains(" " + dataSource.getKrbRealm().toLowerCase() + " = {");
+            if (!isRealmDefined) {
+                confStr = addNewRealm(dataSource, confStr);
+            }
+
+            //remove inconsistent part (if it exists) with old values of pair domain-realm
+            confStr = removeInconsistentDomainRealm(dataSource, confStr);
+            boolean isDomainDefined = confStr.toLowerCase().contains(" " + dataSource.getKrbFQDN() + " = ");
+            if (!isDomainDefined) {
+                confStr = addNewDomainRealm(dataSource, confStr);
+            }
+
+            if (!isRealmDefined || !isDomainDefined) {
+                confStr = confStr.trim();
+                int confLength = confStr.length();
+                raf.seek(0);
+                raf.write(confStr.getBytes());
+                raf.setLength(confLength);
+            }
+        }
+        removeEmptyLines(configPath);
+        return configPath;
+    }
+
+    private void removeEmptyLines(Path configPath) throws IOException {
```

Comment:
synchronized for non-isolated environment to avoid dangerous shared file rewriting